ui/discover/src/main/java/app/tivi/home/discover/Discover.kt

```diff
@@ -76,82 +75,80 @@ import app.tivi.common.compose.bodyWidth
 import app.tivi.common.compose.ui.AutoSizedCircularProgressIndicator
 import app.tivi.common.compose.ui.PosterCard
 import app.tivi.common.compose.ui.TiviStandardAppBar
-import app.tivi.common.compose.viewModel
 import app.tivi.common.ui.resources.MR
 import app.tivi.data.compoundmodels.EntryWithShow
 import app.tivi.data.models.Episode
 import app.tivi.data.models.Season
 import app.tivi.data.models.TiviShow
 import app.tivi.data.traktauth.TraktAuthState
+import app.tivi.overlays.showInDialog
+import app.tivi.screens.AccountScreen
+import app.tivi.screens.DiscoverScreen
+import com.slack.circuit.overlay.LocalOverlayHost
+import com.slack.circuit.runtime.CircuitContext
+import com.slack.circuit.runtime.Screen
+import com.slack.circuit.runtime.ui.Ui
+import com.slack.circuit.runtime.ui.ui
 import dev.icerock.moko.resources.compose.stringResource
-import me.tatarka.inject.annotations.Assisted
+import kotlinx.coroutines.launch
 import me.tatarka.inject.annotations.Inject
 
-/**
- * The type which is used for injecting the function
- */
-typealias Discover = @Composable (
-    openTrendingShows: () -> Unit,
-    openPopularShows: () -> Unit,
-    openRecommendedShows: () -> Unit,
-    openShowDetails: (showId: Long, seasonId: Long?, episodeId: Long?) -> Unit,
-    openUser: () -> Unit,
-) -> Unit
-
 @Inject
-@Composable
-fun Discover(
-    viewModelFactory: () -> DiscoverViewModel,
-    @Assisted openTrendingShows: () -> Unit,
-    @Assisted openPopularShows: () -> Unit,
-    @Assisted openRecommendedShows: () -> Unit,
-    @Assisted openShowDetails: (showId: Long, seasonId: Long?, episodeId: Long?) -> Unit,
-    @Assisted openUser: () -> Unit,
-) {
-    Discover(
-        viewModel = viewModel(factory = viewModelFactory),
-        openTrendingShows = openTrendingShows,
-        openPopularShows = openPopularShows,
-        openRecommendedShows = openRecommendedShows,
-        openShowDetails = openShowDetails,
-        openUser = openUser,
-    )
+class DiscoverUiFactory : Ui.Factory {
+    override fun create(screen: Screen, context: CircuitContext): Ui<*>? = when (screen) {
+        is DiscoverScreen -> {
+            ui<DiscoverUiState> { state, modifier ->
+                Discover(state, modifier)
+            }
+        }
+
+        else -> null
+    }
 }
 
 @Composable
 internal fun Discover(
-    viewModel: DiscoverViewModel,
-    openTrendingShows: () -> Unit,
-    openPopularShows: () -> Unit,
-    openRecommendedShows: () -> Unit,
-    openShowDetails: (showId: Long, seasonId: Long?, episodeId: Long?) -> Unit,
-    openUser: () -> Unit,
+    state: DiscoverUiState,
+    modifier: Modifier = Modifier,
 ) {
-    val viewState = viewModel.presenter()
+    val scope = rememberCoroutineScope()
```

Comment:
You may wanna check out StableCoroutineScope: https://github.com/slackhq/circuit/blob/main/samples/star/src/main/kotlin/com/slack/circuit/star/ui/StableCoroutineScope.kt